lib/command/src/Obelisk/Command/Thunk.hs

```diff
@@ -621,6 +624,33 @@ packThunk' noTrail thunkDir = checkThunkDirectory thunkDir >> readThunk thunkDir
       liftIO $ createThunk thunkDir thunkPtr
       pure thunkPtr
 
+setThunk :: MonadObelisk m => String -> FilePath -> m ()
+setThunk = setThunk' False
+
+setThunk' :: MonadObelisk m => Bool -> String -> FilePath -> m ()
+setThunk' noTrail branch thunkDir = checkThunkDirectory thunkDir >> readThunk thunkDir >>= \case
+  Left err -> failWith $ T.pack $ "thunk set : " <> show err
+  Right (ThunkData_Packed thunkptr) -> do
+    let thunkSrc = _thunkPtr_source thunkptr
+        repository = case thunkSrc of
+          ThunkSource_Git gitSource -> T.unpack $ render $ _gitSource_url gitSource
+          ThunkSource_GitHub gitHubSource -> T.unpack $ render $ _gitSource_url $ forgetGithub True gitHubSource
+    -- check to see if branch exists before modifying directory
+    (exitCode, _) <- gitLsRemoteExitCode repository branch
+    case exitCode of
+      ExitSuccess -> do
+        _ <- unpackThunk' noTrail thunkDir
+        callProcessAndLogOutput (Debug, Error) $ gitProc thunkDir ["checkout", branch]
+        _ <- packThunk' noTrail thunkDir
+        updateThunkToLatest thunkDir
+      ExitFailure errNum -> failWith $ T.pack $ if errNum == 2
+        then "Error: branch not found"
+        else ("Error Code: " <> show errNum <> "issue checking out the desired branch")
+  Right (ThunkData_Checkout _) -> do
+    callProcessAndLogOutput (Debug, Error) $ gitProc thunkDir ["checkout", branch]
```

Comment:
I'm not sure about supporting unpacked thunks for the `thunk set` command. Thoughts @ryantrinkle ?

Reply:
Agreed; I don't see a reason to support it, and it seems dangerous.